server/src/main/java/org/elasticsearch/cluster/routing/allocation/AllocationService.java

```diff
@@ -413,15 +427,43 @@ private void reroute(RoutingAllocation allocation) {
         assert hasDeadNodes(allocation) == false : "dead nodes should be explicitly cleaned up. See disassociateDeadNodes";
         assert AutoExpandReplicas.getAutoExpandReplicaChanges(allocation.metaData(), allocation).isEmpty() :
             "auto-expand replicas out of sync with number of nodes in the cluster";
+        assert assertInitialized();
 
         removeDelayMarkers(allocation);
-        // try to allocate existing shard copies first
-        gatewayAllocator.allocateUnassigned(allocation);
 
+        allocateExistingShards(allocation);  // try to allocate existing shard copies first
         shardsAllocator.allocate(allocation);
         assert RoutingNodes.assertShardStats(allocation.routingNodes());
     }
 
+    private void allocateExistingShards(RoutingAllocation allocation) {
```

Comment:
nit: allocateExistingUnassignedShards?

Reply:
Ok, done in 3301a2b8444c5a37c263754033109d8bee054963.